druid-shell/src/platform/x11/keycodes.rs

```diff
@@ -0,0 +1,335 @@
+            KeyCode::KeyF => f | F,
+            KeyCode::KeyG => g | G,
+            KeyCode::KeyH => h | H,
+            KeyCode::KeyJ => j | J,
+            KeyCode::KeyK => k | K,
+            KeyCode::KeyL => l | L,
+            KeyCode::Semicolon => semicolon,
+            KeyCode::Quote => quoteright,
+            KeyCode::Backslash => backslash,
+
+            KeyCode::KeyZ => z | Z,
+            KeyCode::KeyX => x | X,
+            KeyCode::KeyC => c | C,
+            KeyCode::KeyV => v | V,
+            KeyCode::KeyB => b | B,
+            KeyCode::KeyN => n | N,
+            KeyCode::KeyM => m | M,
+            KeyCode::Comma => comma,
+            KeyCode::Period => period,
+            KeyCode::Slash => slash,
+
+            KeyCode::LeftControl => Control_L,
+            KeyCode::RightControl => Control_R,
+            KeyCode::LeftAlt => Alt_L,
+            KeyCode::RightAlt => Alt_R,
+            KeyCode::LeftShift => Shift_L,
+            KeyCode::RightShift => Shift_R,
+            KeyCode::LeftMeta => Super_L,
+            KeyCode::RightMeta => Super_R,
+
+            KeyCode::Space => space,
+            KeyCode::CapsLock => Caps_Lock,
+            KeyCode::F1 => F1,
+            KeyCode::F2 => F2,
+            KeyCode::F3 => F3,
+            KeyCode::F4 => F4,
+            KeyCode::F5 => F5,
+            KeyCode::F6 => F6,
+            KeyCode::F7 => F7,
+            KeyCode::F8 => F8,
+            KeyCode::F9 => F9,
+            KeyCode::F10 => F10,
+            KeyCode::F11 => F11,
+            KeyCode::F12 => F12,
+
+            KeyCode::PrintScreen => Print,
+            KeyCode::ScrollLock => Scroll_Lock,
+            // Pause/Break not audio.
+            KeyCode::Pause => Pause,
+
+            KeyCode::Insert => Insert,
+            KeyCode::Delete => Delete,
+            KeyCode::Home => Home,
+            KeyCode::End => End,
+            KeyCode::PageUp => Page_Up,
+            KeyCode::PageDown => Page_Down,
+
+            KeyCode::Numpad0 => KP_0,
+            KeyCode::Numpad1 => KP_1,
+            KeyCode::Numpad2 => KP_2,
+            KeyCode::Numpad3 => KP_3,
+            KeyCode::Numpad4 => KP_4,
+            KeyCode::Numpad5 => KP_5,
+            KeyCode::Numpad6 => KP_6,
+            KeyCode::Numpad7 => KP_7,
+            KeyCode::Numpad8 => KP_8,
+            KeyCode::Numpad9 => KP_9,
+
+            KeyCode::NumpadEquals => KP_Equal,
+            KeyCode::NumpadSubtract => KP_Subtract,
+            KeyCode::NumpadAdd => KP_Add,
+            KeyCode::NumpadDecimal => KP_Decimal,
+            KeyCode::NumpadMultiply => KP_Multiply,
+            KeyCode::NumpadDivide => KP_Divide,
+            KeyCode::NumLock => Num_Lock,
+            KeyCode::NumpadEnter => KP_Enter,
+
+            KeyCode::ArrowUp => Up,
+            KeyCode::ArrowDown => Down,
+            KeyCode::ArrowLeft => Left,
+            KeyCode::ArrowRight => Right,
+            KeyCode::Unknown(_) => unreachable!(
```

Comment:
The intention here is that the value of the variant *is* the raw key code for this platform, and you should just return it directly.

Reply:
This one I was able to handle myself ;)